Plugins/Detray/src/DetrayMaterialConverter.cpp

```diff
@@ -248,39 +256,49 @@ Acts::DetrayMaterialConverter::convertGridSurfaceMaterial(
     // Per volume surface selector
     MaterialSurfaceSelector selector;
     volume->visitSurfaces(selector);
-    ACTS_DEBUG("DetrayMaterialConverter: found "
-               << selector.surfaces.size()
-               << " surfaces/portals with material in volume "
-               << volume->name());
+    ACTS_VERBOSE("DetrayMaterialConverter: found "
+                 << selector.surfaces.size()
+                 << " surfaces/portals with material in volume "
+                 << volume->name());
     // Find the voluem index first
-    auto volumeIndex = geoIdCache.volumeLinks.find(volume->geometryId());
-    if (volumeIndex != geoIdCache.volumeLinks.end()) {
+    auto volumeIndex = cCache.volumeLinks.find(volume->geometryId());
+    if (volumeIndex != cCache.volumeLinks.end()) {
       std::vector<DetrayMaterialGrid> volumeMaterialGrids = {};
       // Now convert the surfaces
       for (const auto& surface : selector.surfaces) {
-        // Find the surface index
-        auto surfaceIndices =
-            geoIdCache.localSurfaceLinks.equal_range(surface->geometryId());
-        DetrayMaterialGrid materialGrid =
-            convertGridSurfaceMaterial(*surface->surfaceMaterial(), logger);
-        // Ignore if an empty payload is returned
-        if (materialGrid.axes.empty() && materialGrid.bins.empty()) {
-          continue;
-        }
-        // Loop over the equal range and fill one grid each, this is needed
-        // as the initial portal could be split into multiple surfaces
-        for (auto itr = surfaceIndices.first; itr != surfaceIndices.second;
-             ++itr) {
-          // Fill the surface index
-          materialGrid.owner_link =
-              detray::io::single_link_payload{itr->second};
-          // Fill the grid
-          volumeMaterialGrids.push_back(materialGrid);
+        // Find the surfaces to assign
+        auto vIndex = cCache.volumeIndex(volume);
+        auto localSurfaceLinks = cCache.localSurfaceLinks.find(vIndex);
+        if (localSurfaceLinks != cCache.localSurfaceLinks.end()) {
+          // Find the surface link
+          auto surfaceIndices =
+              localSurfaceLinks->second.equal_range(surface->geometryId());
+
+          ACTS_VERBOSE(
+              "DetrayMaterialConverter: assigning to "
+              << std::distance(surfaceIndices.first, surfaceIndices.second)
+              << " surfaces with material in volume " << volume->name());
+          DetrayMaterialGrid materialGrid =
+              convertGridSurfaceMaterial(*surface->surfaceMaterial(), logger);
+          // Ignore if an empty payload is returned
+          if (materialGrid.axes.empty() && materialGrid.bins.empty()) {
```

Comment:
Should this not be an "or" actually?
```suggestion
          if (materialGrid.axes.empty() || materialGrid.bins.empty()) {
```